packages/api/src/makeServices.ts

```diff
@@ -1,9 +1,46 @@
-import { MakeServices } from './types'
+import { getConfig } from '@redwoodjs/internal'
+
+import { BeforeResolverSpec, MissingBeforeResolver } from './beforeResolverSpec'
+import { ServicesCollection, MakeServices, Services } from './types'
 
-/**
- * This function is a stub for when we fully introduce the concept of
- * services.
- */
 export const makeServices: MakeServices = ({ services }) => {
-  return services
+  if (!getConfig().api.experimentalSecureServices) {
+    console.warn(
+      'NOTICE: Redwood v1.0 will make services secure by default. To optin to this behavior now, add `experimentalSecureServices = true` to the [api] section of your redwood.toml file. For more information: https://redwoodjs.com/docs/secure-services'
+    )
+    return services
+  }
+
+  const servicesCollection: ServicesCollection = {}
+
+  for (const [name, resolvers] of Object.entries(services)) {
+    if (!resolvers?.beforeResolver) {
+      throw new MissingBeforeResolver(name)
```

Comment:
I think it's helpful to append Error to Errors :)
```suggestion
      throw new MissingBeforeResolverError(name)
```

Reply:
Fixed!